src/Shared/Microsoft.DotNet.Internal.Testing.Utility/GenericReadOnlyDictionaryAssertions.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using FluentAssertions.Collections;
+
+namespace Microsoft.DotNet.Internal.Testing.Utility
+{
+    public class GenericReadOnlyDictionaryAssertions<TKey, TValue> : GenericDictionaryAssertions<TKey, TValue>
+    {
+        private class ReadOnlyDictionaryWrapper : IDictionary<TKey, TValue>
+        {
+            private readonly IReadOnlyDictionary<TKey, TValue> _backing;
+
+            public ReadOnlyDictionaryWrapper(IReadOnlyDictionary<TKey, TValue> backing)
+            {
+                _backing = backing;
+            }
+
+            public IEnumerator<KeyValuePair<TKey, TValue>> GetEnumerator()
+            {
+                return _backing.GetEnumerator();
+            }
+
+            IEnumerator IEnumerable.GetEnumerator()
+            {
+                return ((IEnumerable) _backing).GetEnumerator();
+            }
+
+            public void Add(KeyValuePair<TKey, TValue> item)
+            {
+                throw new NotSupportedException();
+            }
+
+            public void Clear()
+            {
+                throw new NotSupportedException();
+            }
+
+            public bool Contains(KeyValuePair<TKey, TValue> item)
+            {
+                throw new NotSupportedException();
+            }
+
+            public void CopyTo(KeyValuePair<TKey, TValue>[] array, int arrayIndex)
+            {
+                throw new NotSupportedException();
```

Comment:
>throw new NotSupportedException(); [](start = 16, length = 34)

This is also fine on Read Only stuff

Reply:
Anything that wasn't on the "IReadOnly" dictionary, I didn't implement.  I could implement this one, I suppose. It's possible something in FluentAssertions is going to care.

Reply:
Made work in a... roundabout way... by calling ToList(). :-)